test/integration/vue.spec.js

```diff
@@ -1,71 +1,83 @@
-
 var describe = require('mocha').describe;
 var it = require('mocha').it;
 var expect = require('chai').expect;
 var path = require('path');
 var webpack = require('webpack');
 var process = require('process');
 var ForkTsCheckerWebpackPlugin = require('../../lib/index');
-var IncrementalChecker = require('../../lib/IncrementalChecker').IncrementalChecker;
+var IncrementalChecker = require('../../lib/IncrementalChecker')
+  .IncrementalChecker;
 
 var webpackMajorVersion = require('./webpackVersion')();
 
-var VueLoaderPlugin = webpackMajorVersion >= 4 ? require('vue-loader/lib/plugin') : undefined;
+var VueLoaderPlugin =
+    webpackMajorVersion >= 4 ? require('vue-loader/lib/plugin') : undefined;
 
-describe('[INTEGRATION] vue', function () {
+describe('[INTEGRATION] vue', function() {
   this.timeout(60000);
-  process.setMaxListeners(0);   
+  process.setMaxListeners(0);
   var plugin;
   var files;
   var compiler;
   var checker;
 
   function createCompiler(options) {
-    plugin = new ForkTsCheckerWebpackPlugin(Object.assign({}, options, { silent: true }));
-
-    compiler = webpack({
-      ...(webpackMajorVersion >= 4 ? { mode: 'development' } : {}),
-      context: path.resolve(__dirname, './vue'),
-      entry: './src/index.ts',
-      output: {
-        path: path.resolve(__dirname, '../../tmp')
-      },
-      resolve: {
-        extensions: ['.ts', '.js', '.vue', '.json'],
-        alias: {
-          '@': path.resolve(__dirname, './vue/src'),
-        }
-      },
-      module: {
-        rules: [
-          {
-            test: /\.vue$/,
-            loader: 'vue-loader'
+    plugin = new ForkTsCheckerWebpackPlugin(
+      Object.assign({}, options, { silent: true })
+    );
+    compiler = webpack(
+      Object.assign(
```

Comment:
why the switch to `Object.assign`?  For my money that makes the code a little "noisier" 😄

Reply:
Object spread is not supported in ES6 syntax, it's ES2018+. Linting with ESLint will fail, and tests might not run with Node 6.

https://node.green/#ES2018-features-object-rest-spread-properties

Reply:
Shouldn't that mean they'd be failing on this already though?

Reply:
Nope! Travis CI is only configured to be running on Node 8 😛